File: Binary Tree - BFS/1161. Maximum Level Sum of a Binary Tree/solution.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function maxLevelSum(root: TreeNode | null): number {
    if (!root) return 0;

    const queue: TreeNode[] = [root];
    let maxSum = -Infinity;
    let level = 1;
    let maxLevel = 1;

    while (queue.length) {
        const levelSize = queue.length;
        let levelSum = 0;

        for (let i = 0; i < levelSize; i++) {
            const node = queue.shift()!;
            levelSum += node.val;

            if (node.left) queue.push(node.left);
            if (node.right) queue.push(node.right);
        }

        if (levelSum > maxSum) {
            maxSum = levelSum;
            maxLevel = level;
        }

        level++;
    }

    return maxLevel;
}

File: Binary Tree - BFS/1161. Maximum Level Sum of a Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
   def maxLevelSum(self, root: Optional[TreeNode]) -> int:
       if not root:
           return 0

       queue = deque([root])
       max_sum = float('-inf')
       level = 1
       max_level = 1

       while queue:
           level_sum = 0
           for _ in range(len(queue)):
               node = queue.popleft()
               level_sum += node.val
               if node.left:
                   queue.append(node.left)
               if node.right:
                   queue.append(node.right)

           if level_sum > max_sum:
               max_sum = level_sum
               max_level = level

           level += 1

       return max_level